src/swarm/neo/protocol/socket/uio_const.d

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+
+    D-`const`-correct vector/gather socket output with optional `SIGPIPE`
+    suppression.
+
+    Copyright: Copyright (c) 2015-2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.protocol.socket.uio_const;
+
+import core.sys.posix.sys.types: ssize_t;
+
+import core.sys.posix.sys.socket;
+
+import ocean.transition;
+
+/// D-`const`-correct output version of `iovec`.
+struct iovec_const
+{
+    Const!(void)* iov_base;
+    size_t iov_len;
+}
+
+/*******************************************************************************
+
+    This is a `const`-correct hybrid of `send(2)` and `writev(2)`:
+    - Like `send` it works only with a connection-mode socket and accepts the
+      same flags as `send`.
+    - Like `writev` it performs gather-output, accepting a vector (i.e. array)
+      of output buffers, which is, in contrast to `writev(2)`,
+      D-`const`-correct.
+
+    Params:
+        socket_fd = the socket file descriptor
+        data      = the output data, see the `writev(2)` docu for details
+        flags     = `MSG_*` flags; see `send(2)` documentation
+
+    Returns:
+        0 on success or -1 on failure, in which case `errno` is set accordingly.
+
+*******************************************************************************/
+
+ssize_t sendv ( int socket_fd, in iovec_const[] data, int flags = 0 )
+{
+    Const!(msghdr) m = {
+            msg_iov:    cast(Const!(iovec*))data.ptr,
+            msg_iovlen: data.length
+    };
+    return sendmsg(socket_fd, &m, flags);
+}
+
+// Ensure binary compatibility of `iovec_const` with `iovec`.
+unittest
```

Comment:
Would be cool to have a template in ocean: `BinaryLayoutIdentical`, or some such.